library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -316,11 +316,21 @@ return_status libspdm_set_data(IN void *context, IN libspdm_data_type_t data_typ
         if (data_size > LIBSPDM_MAX_CERT_CHAIN_SIZE) {
             return RETURN_OUT_OF_RESOURCES;
         }
-        spdm_context->connection_info.peer_used_cert_chain_buffer_size =
-            data_size;
-        copy_mem(spdm_context->connection_info
-                 .peer_used_cert_chain_buffer,
-             data, data_size);
+        libspdm_hash_all(
+            spdm_context->connection_info.algorithm.base_hash_algo,
+            data, data_size,
+            spdm_context->connection_info.peer_used_cert_chain_buffer_hash);
+
+        spdm_context->connection_info.peer_used_cert_chain_buffer_hash_size =
+            libspdm_get_hash_size(spdm_context->connection_info.algorithm.base_hash_algo);
+        libspdm_get_public_key_from_cert_chain(spdm_context,
+            spdm_context->connection_info.algorithm.base_asym_algo,
+            data, data_size,
+            &spdm_context->connection_info.public_key_with_base_asym_algo);
+        libspdm_get_public_key_from_cert_chain(spdm_context,
```

Comment:
Need error handling here.

Why we need both base_asym_algo and req_base_asym_alg ? Should we only use one of them?
E.g. requester only uses base_asym_algo, and responder only uses req_base_asym_alg.